include/phasar/DataFlow/IfdsIde/Solver/FlowFunctionCache.h

```diff
@@ -0,0 +1,353 @@
+        });
+  }
+
+  decltype(auto) getCallFlowFunction(ProblemTy &Problem,
+                                     ByConstRef<n_t> CallSite,
+                                     ByConstRef<f_t> Callee,
+                                     uint64_t CSCalleeId) {
+    return getFlowFunction<call_ff_t>(CallFFCache, CSCalleeId, Problem,
+                                      [CallSite, Callee](ProblemTy &Problem) {
+                                        return Problem.getCallFlowFunction(
+                                            CallSite, Callee);
+                                      });
+  }
+
+  /// CAUTION: Keep the ID definitions in mind!
+  decltype(auto) getRetFlowFunction(ProblemTy &Problem,
+                                    ByConstRef<n_t> CallSite,
+                                    ByConstRef<f_t> Callee,
+                                    ByConstRef<n_t> ExitInst,
+                                    ByConstRef<n_t> RetSite, uint64_t CSExitId,
+                                    uint64_t CalleeRSId) {
+    if constexpr (needs_cache_v<ret_ff_t>) {
+      if constexpr (std::is_base_of_v<
+                        llvm::Instruction,
+                        std::remove_cv_t<std::remove_pointer_t<n_t>>>) {
+        if (llvm::isa<llvm::CallInst>(CallSite)) {
+          auto &Ret = SimpleRetFFCache.getOrCreate(CSExitId);
+          if (!Ret) {
+            Ret =
+                Problem.getRetFlowFunction(CallSite, Callee, ExitInst, RetSite);
+          }
+          return wrapFlowFunction(Problem, Ret);
+        }
+      }
+
+      auto &Ret = RetFFCache.getOrCreate(std::make_pair(CSExitId, CalleeRSId));
+      if (!Ret) {
+        Ret = Problem.getRetFlowFunction(CallSite, Callee, ExitInst, RetSite);
+      }
+      return wrapFlowFunction(Problem, Ret);
+    } else {
+      return wrapFlowFunction(
+          Problem,
+          Problem.getRetFlowFunction(CallSite, Callee, ExitInst, RetSite));
+    }
+  }
+
+  decltype(auto) getSummaryFlowFunction(ProblemTy &Problem,
+                                        ByConstRef<n_t> CallSite,
+                                        ByConstRef<f_t> Callee,
+                                        uint64_t /*CSCalleeId*/) {
+    if constexpr (needs_cache_v<summary_ff_t>) {
+      return GenericFlowFunction<d_t>(
+          Problem.getSummaryFlowFunction(CallSite, Callee));
+    } else {
+      return Problem.getSummaryFlowFunction(CallSite, Callee);
+    }
+
+    /// XXX: The old FECache doesn't cache summary FFs, so refrain from doing
+    /// this here as well. Enable it, once the user-problems reliably return
+    /// std::nullptr_t such that caching can be disabled automatically
+
+    // return getFlowFunction<summary_ff_t>(
+    //     SummaryFFCache, CSCalleeId, Problem,
+    //     [CallSite, Callee](ProblemTy &Problem) {
+    //       return Problem.getSummaryFlowFunction(CallSite, Callee);
+    //     });
```

Comment:
Left over?

Reply:
I left this code here, such that you can easily turn it on, once needed. Probably, this will never happen, so removed it.